test/ecto/multi_test.exs

```diff
@@ -196,6 +196,18 @@ defmodule Ecto.MultiTest do
     assert [{:comments, {:insert_all, Comment, [[x: 2]], []}}] = multi.operations
   end
 
+  test "insert_all fun" do
+    fun_schema = fn _changes -> {:ok, Comment} end
+    fun_entries = fn _changes -> {:ok, [[x: 2]]} end
```

Comment:
It should return just a list, no? I wonder if we can have a more precise test, because this one and the ones below are not catching much. :)

Reply:
> It should return just a list, no? 

I think it is correct. We have a list of entries, each entry can be a keyword list as `[x: 2]`

> I wonder if we can have a more precise test, because this one and the ones below are not catching much. :)

Yes, the scenario is clearly wrong, but the tests are still passing. WDYT of running the multi with `TestRepo.transaction` to have more cover of the functionality?

Reply:
> Yes, the scenario is clearly wrong, but the tests are still passing. WDYT of running the multi with TestRepo.transaction to have more cover of the functionality?

Sounds good!

Reply:
> It should return just a list, no?

I had misunderstood. You were right. Most of the examples in the added tests had this same problem.

> Sounds good!

I just updated the tests to run the multi. Could you take another look?